x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ssl/StoreKeyConfig.java

```diff
@@ -78,6 +78,18 @@ X509ExtendedKeyManager createKeyManager(@Nullable Environment environment) {
         Path ksPath = keyStorePath == null ? null : CertParsingUtils.resolvePath(keyStorePath, environment);
         try {
             KeyStore ks = getStore(ksPath, keyStoreType, keyStorePassword);
+            // TBD: filte out only http.ssl.keystore
+            ArrayList<String> aliases = Collections.list(ks.aliases());
+            if (this instanceof StoreKeyConfig && aliases.size() > 1) {
+                for (String alias : aliases) {
+                    Certificate certificate = ks.getCertificate(alias);
```

Comment:
we need to only be doing this for PrivateKeyEntry objects, not generally. This code will be removing TrustedCertificateEntry objects ( which _are_ CA certificates ) from the keystore and we do not want that. You can filter the aliases further with `ks.isKeyEntry(alias)` ( see `getPrivateKeyEntries` below for an example ) and only remove entries if the PrivateKeyEntries are more than 1

Reply:
Added tests